python-lib/dku_stress_test_center/metrics.py

```diff
@@ -0,0 +1,32 @@
+import numpy as np
+from sklearn.metrics import accuracy_score
+
+def get_performance_metric(metric_name: str):
+    return accuracy_score # TODO
+
+def greater_perf_is_better(metric_name: str):
+    return True # TODO
+
+def worst_group_performance(metric_name: str, y_true: np.array, y_pred: np.array):
+    performances = []
+    target_classes = np.unique(y_true)
+    for target_class in target_classes:
+        class_mask = y_true == target_class
+        performance = get_performance_metric(metric_name)(y_true[class_mask], y_pred[class_mask])
+        performances.append(performance)
+    return min(performances)
+
+def stress_resilience(clean_y_pred: np.array, perturbed_y_pred: np.array):
+    # TODO: make it work for regression as well
+    return (clean_y_pred == perturbed_y_pred).sum() / len(clean_y_pred)
+
+def performance_variation(metric_name: str, clean_y_true: np.array, clean_y_pred: np.array,
+                          perturbed_y_true: np.array, perturbed_y_pred: np.array):
+    greater_is_better = greater_perf_is_better(metric_name)
+    performance_metric = get_performance_metric(metric_name)
+
+    clean_performance_metric = performance_metric(clean_y_true, clean_y_pred)
+    stressed_performance_metric = performance_metric(perturbed_y_true, perturbed_y_pred)
+
+    delta = stressed_performance_metric - clean_performance_metric
+    return delta if greater_is_better else - delta # TODO might be trickier than this
```

Comment:
Yes, I think it can work like this, if we keep the value in their true unit (not a percentage). I'm thinking a MAPE going from 0.01 to 0.05, we display perf drop: -0.04. You will see your MAPE degraded of 4 pts. Makes sense to me.

Reply:
So from what I understand:

### Classification
- `accuracy`, `precision`, `recall`, `f1`, `cost matrix gain`, `roc auc` & `lift` : **greater_is_better**; use `stressed_performance_metric - clean_performance_metric`
- `log loss` & `calibration loss` : **lower is better**; use `clean_performance_metric - stressed_performance_metric`

### Regression
- `evs` & `r2 score` : **greater is better**
- `mape`,`mae`, `mse`, `rmse`, `rmsle` : **lower is better**
    - [ ] Question: So we can still use `clean_performance_metric - stressed_performance_metric` here? Even when we have formulas involving non linear operators like logs, powers or roots?

:warning: For `correlation` I am not sure. IIUC correlation is between -1 and 1, and the further from 0 the better.
    - [ ] Question:  do we want to use the absolute value of the correlation before doing the subtraction (and then assume greater is better, so we take `stressed_performance_metric - clean_performance_metric`)? Like:
         1. If we have .6 for clean and .3 for perturbed, variation is -.3
         2. If we have - .6 for clean and -.3 for perturbed, variation is also -.3
         3. If we have .6 for clean (resp -.6) and -.3 (resp .3) for perturbed), variation is still -.3

This can be also discussed further on the channel/in a meeting of course.

Reply:
Classification: all correct
Regression: 
   * yes, I think we should still use the difference, regardless the formula of the metrics
   * yes for correlation, what you propose is the only way it makes sense to me